compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -67,6 +68,9 @@ object Implicits {
     final val Extension = 4
   }
 
+  val testOneImplicitTimeOut = 1000
+  val suggestImplicitTimeOut = 10000
```

Comment:
The problem is that it can take so long that you think your compiler froze. I think it's better to be safe here. Another thing we should watch is memory. Not good if the compiler does with an OutOfMemory exception here. I am not sure how that is best done. But it could also come later.